### DISCUSSION TITLE
Scratch Addons 2 Megathread

### DISCUSSION THREAD

**Discussion by Hans5958** [ScratchAddons/ScratchAddons#5103] [Discussion]
This discussion is for changes that should be in a future major version of Scratch Addons (dubbed Scratch Addons 2.0). There is still no concrete plan for this future version, and still in discussion.

It is preferred to have one idea in one comment, just so we can discuss the specific idea cleanly in one reply section (or on the linked discussion/issue). But, go crazy.

This is a continuation on #5001, just because that is just a discussion and not a plan, and so we can have the goodies of a discussion post.

cc @WorldLanguages @lisa-wolfgang @RedGuy12 @GrahamSH-LLK @apple502j 

# Contents

- [Build step](https://github.com/ScratchAddons/ScratchAddons/discussions/5103#discussioncomment-3630986)
- [Redesign](https://github.com/ScratchAddons/ScratchAddons/discussions/5103#discussioncomment-3631004)
- [Manifest v3](https://github.com/ScratchAddons/ScratchAddons/discussions/5103#discussioncomment-3631008)

**Comment by Hans5958** [comment]
# Build step

Building the extension, instead of simple packaging.

- #1301 
- #4886

**Comment by Hans5958** [comment]
# Redesign

The extensive redesign of the extension. Mainly focused on the settings page, but may also include other parts of the extension.

Please make each prototype/mockup in different discussion post! It would make it easier to navigate it here.

- #4991
- #5864
- #5893 
- #6317
- #7215 #7214

**Reply by iqnite** [reply]
Please add the ability to pin specific addons to the top of the page!

**Reply by WorldLanguages** [reply]
@iqnite I think there's an issue about that already

**Reply by BroJac5246** [reply]
> @iqnite I think there's an issue about that already

I randomly stumbled upon it just now: #1250

**Reply by DNin01** [reply]
It would be great to collect feedback from collaborators and, if possible, users before focusing on a new design. Should we start a discussion just for that? We could use it to let everyone propose and provide feedback on new designs.

**Reply by Hans5958** [reply]
#5893 has been added.

**Reply by Secret-chest** [reply]
> It would be great to collect feedback from collaborators and, if possible, users before focusing on a new design. Should we start a discussion just for that? We could use it to let everyone propose and provide feedback on new designs.

Maybe each design should use a separate thread

**Reply by Hans5958** [reply]
> Maybe each design should use a separate thread

Separate posts would be better

**Reply by Hans5958** [reply]
#6317 has been added.

**Reply by Hans5958** [reply]
#5864 has been added. I just found this out lately.

**Reply by Hans5958** [reply]
#7215 and #7214 has been added as a single entry.

**Reply by Waakul** [reply]
i highly recommend using my redesign cause other ones are just some cringey (no offence) stuff people made for fun...

**Reply by Waakul** [reply]
also mine is also implemented in a pr (draft for now)

**Comment by Hans5958** [comment]
# Welcoming Manifest v3 (curse you Google!)

(some say that it wouldn't be on the SA2 because of the deadline)

- #1009
- https://github.com/ScratchAddons/ScratchAddons/milestone/13
- #5162 
- #6876

**Reply by Samq64** [reply]
Firefox now supports MV3: https://www.mozilla.org/en-US/firefox/109.0/releasenotes/

**Reply by WorldLanguages** [reply]
> We will provide sufficient migration time for developers - at least 6 months of heads-up - before beginning any experiments to turn off MV2 in the browser next year [2024].

https://groups.google.com/u/1/a/chromium.org/g/chromium-extensions/c/zQ77HkGmK9E (March 29th 2023)

**Reply by WorldLanguages** [reply]
See updated dates: https://github.com/ScratchAddons/ScratchAddons/discussions/5162#discussioncomment-7591611

**Comment by BroJac5246** [comment]


**Reply by WorldLanguages** [reply]
Could be

**Reply by Hans5958** [reply]
See https://github.com/ScratchAddons/ScratchAddons/discussions/5103#discussioncomment-3631004.

**Comment by Samq64** [comment]
Firefox 109 supports MV3 and will be pushed out to users tomorrow.

**Reply by Hans5958** [reply]
Comment on https://github.com/ScratchAddons/ScratchAddons/discussions/5103#discussioncomment-3631008, please.

**Comment by Waakul** [comment]


**Reply by Waakul** [reply]
it'd be cool to have this major change in v2

**Reply by BroJac5246** [reply]
There's no reason not to introduce this as soon as it's ready. It doesn't introduce any new user-facing changes other than drastically reducing load times.

**Reply by Waakul** [reply]
idk but they already implemented.

**Reply by BroJac5246** [reply]
> idk but they already implemented.

Implemented what?

**Reply by Waakul** [reply]
see the pr, it's implemented there

**Reply by BroJac5246** [reply]
> see the pr, it's implemented there

I'm not saying it shouldn't be merged at all, but rather that it should be merged as soon as possible. Preparing a 2.0 version of the extension would take many months of planning, coding, and testing, but Vue3 is almost ready and is super helpful while not being something that users will perceive as a new feature.

**Reply by Waakul** [reply]
good point
